third-party/sysdeps/common/zstd/patch_source.sh

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/bash
+set -e
+
+SOURCE_DIR="${1:?Source directory must be given}"
+ZSTD_CMAKELIST="$SOURCE_DIR/build/cmake/lib/CMakeLists.txt"
+echo "Patching sources..."
+
+sed -i -E 's/(OUTPUT_NAME)[[:space:]]+zstd/\1 rocm_sysdeps_zstd/' "$ZSTD_CMAKELIST"
```

Comment:
These `patch_source.sh` files are sketchy. I would much prefer to achieve the same effect through CMake variables with the existing source somehow.

Are other downstream users of zstd and zlib patching its source in similar ways? Should we fork zlib and zstd into git repositories we control instead of tacking on scripts like this in the middle of the build system?

Reply:
> These `patch_source.sh` files are sketchy. I would much prefer to achieve the same effect through CMake variables with the existing source somehow.
> 
> Are other downstream users of zstd and zlib patching its source in similar ways? Should we fork zlib and zstd into git repositories we control instead of tacking on scripts like this in the middle of the build system?

Currently the source codes ( zlib,zstd and elfutils ) are downloaded  while building. 
https://github.com/ROCm/TheRock/blob/bee658b0d7c89e636fa04356711edb5366c8b213/third-party/sysdeps/common/zlib/CMakeLists.txt#L9
As you mentioned if the source codes  for these libraries are hosted in github  , we can avoid patching source files

Reply:
We control those mirrors: https://github.com/ROCm/TheRock/blob/main/docs/development/git_chores.md#updating-a-third-party-mirror

I'm first asking about the source code itself. Code hosting and downloading is secondary. What would it take to fix the issues you noticed without needing to patch the sources? Add a CMake variable that renames a binary/target name? Add a post-build step that does that instead?

Reply:
> We control those mirrors: https://github.com/ROCm/TheRock/blob/main/docs/development/git_chores.md#updating-a-third-party-mirror
> 
> I'm first asking about the source code itself. Code hosting and downloading is secondary. What would it take to fix the issues you noticed without needing to patch the sources? Add a CMake variable that renames a binary/target name? Add a post-build step that does that instead?

Fixing in source code is straightforward. In source code currently target name is  hard coded and we may need to change it to a variable .
But we do see patch_source editing in other modules like elfutils, numactl
https://github.com/ROCm/TheRock/blob/main/third-party/sysdeps/linux/elfutils/patch_source.sh
That is why we have used a similar approach here as well.

Reply:
Ah, okay... if other sysdeps do it then it's fine here. I'll still grumble a bit though :P